packages/angular/src/tracing.ts

```diff
@@ -241,3 +279,44 @@ export function TraceMethodDecorator(): MethodDecorator {
     return descriptor;
   };
 }
+
+/**
+ * Recursively traverses the routerstate and its children to collect a map of parameters on the entire route.
+ *
+ * Because Angular supports child routes (e.g. when creating nested routes or lazy loaded routes), we have
+ * to not only visit the root router snapshot but also its children to get all parameters of the entire route.
+ *
+ * @param activatedRouteSnapshot the root router snapshot
+ * @returns a map of params, mapping from a key to an array of values
+ */
+export function getParamsOfRoute(activatedRouteSnapshot: ActivatedRouteSnapshot): ParamMap {
+  function getParamsOfRouteH(routeSnapshot: ActivatedRouteSnapshot, accParams: ParamMap): ParamMap {
+    Object.keys(routeSnapshot.params).forEach(key => {
+      accParams[key] = [...(accParams[key] || []), routeSnapshot.params[key]];
+    });
+    routeSnapshot.children.forEach(child => getParamsOfRouteH(child, accParams));
+    return accParams;
+  }
+
+  return getParamsOfRouteH(activatedRouteSnapshot, {});
+}
+
+/**
+ * Takes a raw URl and a map of params and replaces each values occuring in the raw URL with
+ * the name of the parameter.
+ *
+ * @param url raw URL (e.g. /user/1234/details)
+ * @param params a map of type ParamMap
+ * @returns the parameterized URL (e.g. /user/:userId/details)
+ */
+export function getParameterizedRouteFromUrlAndParams(url: string, params: ParamMap): string {
+  if (params) {
+    return Object.keys(params).reduce((prevUrl: string, paramName: string) => {
+      return prevUrl
+        .split('/')
+        .map(segment => (params[paramName].find(p => p === segment) ? `:${paramName}` : segment))
+        .join('/');
```

Comment:
Just want to make sure we evaluated the following approach:

I believe (*really only believe and not know*) we have a route config available in `ev.state.root.routeConfig` and all the child snashots. The route config should contain the parameterized route if I understand everything correctly.
Have you tried to stitch the routes together when traversing that tree?

Reply:
I inspected `routeConfig` for some time before I went with the interpolation approach. Interestingly enough, if you explore it, its first child indeed contains a parameterized route but it's not the complete route. The last part is missing for some reason.
Similarly, the url segments in the routerstate identify a segment as a parameter (if it is one) but they do not reference or contain the name of that parameter (e.g. `:id` or similar). 
Anyway, it still bugs me that I can't seem to get the complete route. I'll give this one more hour in the evening to figure out and if by then I don't find something better, I'll stay with the current approach